proxy/certs/certs.go

```diff
@@ -298,6 +298,10 @@ func (s *RemoteCertSource) Remote(instance string) (cert *x509.Certificate, addr
 
 // TokenExpiration returns the expiration time for token source associated with remote cert source.
 func (s *RemoteCertSource) TokenExpiration() (ret time.Time, err error) {
+	// if no token is being used, return zero for expiration
+	if s.TokenSource == nil {
+		return ret, err
```

Comment:
Minor nit: Might be nice to remove the named return values while you're in here. But not a big deal.